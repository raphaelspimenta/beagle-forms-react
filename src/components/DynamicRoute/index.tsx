/* eslint-disable react/jsx-props-no-spreading */
import React, { lazy, memo, Suspense, ComponentType } from 'react'
import Loader from 'components/Loader'
import ErrorBoundary from 'components/ErrorBoundary'

type ImportPath = Promise< { default: ComponentType<any> }>

const DynamicRoute = (importPath: () => ImportPath) => {
  const Component = lazy(importPath)

  const DynamicComponent = (props: any) => (
    <ErrorBoundary>
      <Suspense fallback={<Loader />}>
        <Component {...props} />
      </Suspense>
    </ErrorBoundary>
  )

  return memo(DynamicComponent)
}

export default DynamicRoute
