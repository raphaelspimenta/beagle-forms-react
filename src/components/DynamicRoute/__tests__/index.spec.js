import React from 'react'
import { shallow } from 'enzyme'
import DynamicRoute from '../'

describe('DynamicRoute', () => {

  const Home = DynamicRoute(() => import('containers/Home'/* webpackChunkName: 'home-screen' */))

  it('should render', () => {
    const component = shallow(<Home />)
    expect(component).toMatchSnapshot()
  })
})
