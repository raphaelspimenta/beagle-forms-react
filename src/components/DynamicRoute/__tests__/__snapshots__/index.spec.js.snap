// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicRoute should render 1`] = `
<ErrorBoundary>
  <Suspense
    fallback={<Loader />}
  >
    <lazy />
  </Suspense>
</ErrorBoundary>
`;
